geth/node/manager.go

```diff
@@ -31,86 +34,90 @@ var (
 	ErrRPCClient                   = errors.New("failed to init RPC client")
 )
 
+const (
+	started = 0
+	stopped = 1
+	pending = 2
+)
+
 // NodeManager manages Status node (which abstracts contained geth node)
+// uses interfaces to prevent races on exported objects.
 type NodeManager struct {
-	sync.RWMutex
-	config         *params.NodeConfig // Status node configuration
-	node           *node.Node         // reference to Geth P2P stack/node
-	nodeStarted    chan struct{}      // channel to wait for start up notifications
-	nodeStopped    chan struct{}      // channel to wait for termination notifications
-	whisperService *whisper.Whisper   // reference to Whisper service
-	lesService     *les.LightEthereum // reference to LES service
-	rpcClient      *rpc.Client        // reference to RPC client
+	state *int32
+
+	config atomic.Value // Status node configuration
+
+	node     geth.Node // reference to Geth P2P stack/node
+	nodeLock sync.RWMutex
+
+	nodeStarted     chan struct{} // channel to wait for start up notifications
+	nodeStartedLock sync.RWMutex
```

Comment:
Do we need to guard closing a channel? All these locks can be replaced with a single one, I guess.

Reply:
All this locks has different responsobilities: in code we could easely sow what we lock now and for what.
This channel mutes in needed to reinit NodeManager with new node, which is possible.